google/cloud/bigtable/row_filters.py

```diff
@@ -35,6 +41,21 @@ class RowFilter(object):
         This class is a do-nothing base class for all row filters.
     """
 
+    def _to_pb(self) -> data_v2_pb2.RowFilter:
+        """Converts the row filter to a protobuf.
+
+        Returns: The converted current object.
+        """
+        return data_v2_pb2.RowFilter(**self.to_dict())
+
+    def to_dict(self) -> dict[str, Any]:
+        """Converts the row filter to a dict representation."""
+        # unimplemented on base class
+        raise NotImplementedError
```

Comment:
What do you think of creating an ABC using abc.ABC for `RowFilter` so that `to_dict` can be one of its `@abstractmethod`

Reply:
Yeah I think that makes sense. The only complication is that it requires us to make other filter subclasses abstract as well (_BoolFilter, _FilterCombination, _CellCountFilter), but that's probably good to prevent users from instantiating them directly anyway

I made the changes, let me know what you think